virttest/qemu_vm.py

```diff
@@ -962,8 +962,7 @@ def add_memorys(devices, params):
                     options.append("slots=%s" % mem_params["slots"])
                     options.append("maxmem=%s" % mem_params["maxmem"])
                 for name in mem_params.objects("mem_devs"):
-                    memdev_params = mem_params.object_params(name)
-                    dev = devices.memory_define_by_params(memdev_params, name)
+                    dev = devices.memory_define_by_params(mem_params, name)
```

Comment:
`mem_params` contains params which have dropped suffix `_mem` (at line 926), however in `memory_define_by_params()`, the param `use_mem` is required, so that would be a problem.

Reply:
@luckyh  I don't think so.  The params in cfg are:
```
108             - unused_memory:                                                
109                     target_mem = "unused"                                       
110                     mem_devs += " ${target_mem}"                                
111                     use_mem_unused = no 
```
Param "use_mem_unused" would keep it it until L964,  then it will be "use_mem" when pass to ``memory_define_by_params``.  Thanks!

Reply:
We also have:
```
qemu/tests/cfg/numa_memdev_options.cfg:7:    use_mem = no
```
I am not sure which one is incorrect, but we should have a look about this.

Reply:
@luckyh Both of the cases passed.  I checked `object_params`, it will add new keys to the dict and keep old ones as well.  So in case numa_memdev_options, after L964, "use_mem" and "use" are both in `mem_params`. Thanks!
```
for key in list(new_dict.keys()):
            if key.endswith(suffix):
                new_key = key.split(suffix)[0]
                new_dict[new_key] = new_dict[key]
        return new_dict
```

Reply:
@huangyum after read the whole part of memory related code, now I understand that the problem is the current implementation incorrectly used the same `Params` object to represent `mem` (`-m ..`) and `memdev` (`-object memory-backend-.. -device ..`).

I think that each `memdev` object should be get by `params.object_params()`, not `mem_params.object_params()`, because using `mem_params` will more likely result in name conflict of the parameters.

So I'd suggest to replace `mem_params` to `params` in this line, and correct the name of memdev related parameters in tp-qemu.

Reply:
@luckyh Thank you for so comprehensive review.  I didn't notice that before.  I agree it is better to put `params` here and just updated. 
As to parameters in tp-qemu, seems we don't need to update. No conflict so far, parameters are passed correctly. 
(1/1) Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.hotplug_memory.before.guest_reboot.hotplug.backend_ram.policy_default.two: PASS (317.54 s)

Reply:
I do see some parameters which required a update, such as:
```
./qemu/tests/cfg/numa_memdev_options.cfg:24:            policy_mem = default
                                                              ^^^^
./qemu/tests/cfg/hotplug_mem.cfg:64:            backend_mem = memory-backend-ram
                                                       ^^^^
...
```
can you please double confirm this?

Reply:
@luckyh I don't think they need update, all work well. Thanks!